packages/website/docs/examples/block-from-tiptap.md

```diff
@@ -14,540 +14,379 @@ const { isDark } = useData();
   BlockNoteDOMAttributes,
-  createTipTapBlock,
+  BlockNoteEditor,
+  BlockSchema,
   BlockSpec,
+  blockStyles as styles,
+  mergeCSSClasses,
   PropSchema,
-  BlockSchema,
-  camelToDataKebab,
 } from "@blocknote/core";
-import "@blocknote/core/style.css";
-import { InlineContent } from "@blocknote/react";
-import {
-  NodeViewProps,
-  NodeViewWrapper,
-  ReactNodeViewRenderer,
-} from "@tiptap/react";
-import { Checkbox } from "@mantine/core";
-import { FC, useState } from "react";
-
-// Defines the block's props
-const todoListItemPropSchema = {
-  checked: {
-    default: "false" as const,
-    values: ["true", "false"] as const,
+
+export interface ImageOptions {
+  // Blocks cannot be inline, so this option is not needed.
+  // inline: boolean;
+  allowBase64: boolean;
+  HTMLAttributes: Record<string, any>;
+
+  // Each node you turn into a block should have the `domAttributes` option.
+  domAttributes?: BlockNoteDOMAttributes;
+  // You can also add the `editor` option if you need access to the BlockNote
+  // editor instance. This is marked optional here as the Image node uses
+  // `addOptions`, but the BlockNote instance will always pass itself into
+  // blocks provided in the schema, so you can assume it's always defined.
+  editor?: BlockNoteEditor<
+    BlockSchema & { image: BlockSpec<"image", typeof imagePropSchema> }
+  >;
+}
+
+declare module "@tiptap/core" {
+  interface Commands<ReturnType> {
+    image: {
+      /**
+       * Add an image
+       */
+      setImage: (options: {
+        src: string;
+        alt?: string;
+        title?: string;
+      }) => ReturnType;
+    };
+  }
+}
+
+export const inputRegex =
+  /(?:^|\s)(!\[(.+|:?)]\((\S+)(?:(?:\s+)["'](\S+)["'])?\))$/;
+
+// We define the prop schema here, as we want to make sure the props line up with
+// the attributes we define in `addAttributes` below.
+export const imagePropSchema = {
+  src: {
+    default: "" as const,
+  },
+  alt: {
+    default: "" as const,
+  },
+  title: {
+    default: "" as const,
   },
 } satisfies PropSchema;
 
-// You should define your block using the `createTipTapBlock` function. This
-// takes the same config object as TipTap's `Node.create`, but locks the `group`
-// option to "blockContent" and helps with typing.
-const TodoListItem = createTipTapBlock<
-  "todoListItem",
-  {
-    editor: BlockNoteEditor<
-      BlockSchema & {
-        todoListItem: BlockSpec<"todoListItem", typeof todoListItemPropSchema>;
-      }
-    >;
-    domAttributes?: BlockNoteDOMAttributes;
-  }
->({
-  // TipTap node name should be the same as the block name
-  name: "todoListItem",
-  // Content should be "inline*" or "", depending on if the block should have
-  // inline content (i.e. an editable rich text field). Since we want the block
-  // to have editable text, we make it "inline*".
-  content: "inline*",
-
-  // All props defined in the prop schema should be added to the node's attrs
+export const Image = Node.create<ImageOptions>({
```

Comment:
I still think this example is of very limited use, as it doesn't show how to reuse an existing TipTap extension. We're basically creating an entire TipTap node from scratch here. If that's the case, the user should just create a Custom Block directly right?

Reply:
I mean this is pretty much just the process of converting a node that can be used with any schema to one which is specifically for BlockNote's schema. To make it work as well as the other blocks you need to make all of these changes. I assumed the use case would be more like if someone was migrating from TipTap and had already made a few complex nodes and I think this is where we're miscommunicating.

I think there is some value to creating blocks from TipTap nodes this way, namely native input rules, keyboard shortcuts, PM extensions that you can't do with `createReactBlockSpec`. But there should probably be a more simple example for passing a TipTap extension through `_tiptapOptions`.